plugins/catalog-backend-module-aws/config.d.ts

```diff
@@ -32,5 +32,18 @@ export interface Config {
         };
       };
     };
+    /**
+     * List of provider-specific options and attributes
+     */
+    providers?: {
+      /**
+       * AwsS3EntityProvider / AwsS3EntityProviderFactory configuration
+       */
+      awsS3?: Array<{
+        bucket: string;
+        prefix?: string;
+        region: string;
```

Comment:
initial reaction here:

- is the region necessary? usually the s3 client can deduce it right? unless we HAVE to go via urls.
- i wonder how this relates to the other places that we have config for s3, such as [here](https://github.com/backstage/backstage/blob/949243d991709364647ccdd3acf9625a06179754/plugins/techdocs-backend/config.d.ts#L62) and [here](https://github.com/backstage/backstage/blob/949243d991709364647ccdd3acf9625a06179754/packages/integration/config.d.ts#L210).

It's getting pretty spread out, wondering if there should be ONE place that you configure the actual integration (probably under integrations), and then everything else just "references" those. Maybe that's actually kind of what happens here, but the region parameter threw me off a bit. Also the choice of naming patterns, "bucket" vs "bucketName" etc

Reply:
for S3 bucket, is is required to provide a region as they are region specific and not necessarily in the same region as you are. We could make it optional though and use the default region (e.g., the deployment's region).

Reply:
re bucket vs bucketName: I'm open for both. At other places, bucket is used (e.g., AwsS3UrlReader.ts) or Bucket at the S3 APIs, but there is also bucketName at techdocs.

I can change it to bucketName.

Reply:
I made region optional. It will use the default region as fallback.
And I renamed `bucket` -> `bucketName`.

Reply:
And regarding integrations: It uses the credentials set up under integrations.